File: uikit/src/main/java/io/agora/chat/uikit/chat/adapter/EaseChatExtendMenuAdapter.java
package io.agora.chat.uikit.chat.adapter;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import io.agora.chat.uikit.R;
import io.agora.chat.uikit.adapter.EaseBaseChatExtendMenuAdapter;
import io.agora.chat.uikit.chat.widget.EaseChatExtendMenu;
import io.agora.chat.uikit.chat.widget.EaseChatExtendMenuDialog;
import io.agora.chat.uikit.interfaces.OnItemClickListener;


public class EaseChatExtendMenuAdapter extends EaseBaseChatExtendMenuAdapter<EaseChatExtendMenuAdapter.ViewHolder, EaseChatExtendMenu.ChatMenuItemModel> {
    private boolean isHorizontal;
    private OnItemClickListener itemListener;
    
    public EaseChatExtendMenuAdapter() {}
    
    public EaseChatExtendMenuAdapter(boolean isHorizontal) {
        this.isHorizontal = isHorizontal;
    }

    @Override
    protected int getItemLayoutId() {
        return isHorizontal? R.layout.ease_chat_menu_item_horizontal : R.layout.ease_chat_menu_item;
    }

    @Override
    protected ViewHolder easeCreateViewHolder(View view) {
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        EaseChatExtendMenu.ChatMenuItemModel item = mData.get(position);
        holder.imageView.setBackgroundResource(item.image);
        holder.textView.setText(item.name);
        holder.itemView.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                if(item.clickListener != null){
                    item.clickListener.onChatExtendMenuItemClick(item.id, v);
                }
                if(itemListener != null) {
                    itemListener.onItemClick(v, position);
                }
            }
        });
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.itemListener = listener;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView imageView;
        private TextView textView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = (ImageView) itemView.findViewById(R.id.image);
            textView = (TextView) itemView.findViewById(R.id.text);
        }
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/chat/adapter/EaseChatExtendMenuAdapter.kt
package com.hyphenate.easeui.feature.chat.adapter

import android.content.res.ColorStateList
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.widget.AppCompatImageView
import androidx.recyclerview.widget.RecyclerView
import com.hyphenate.easeui.R
import com.hyphenate.easeui.base.EaseBaseChatExtendMenuAdapter
import com.hyphenate.easeui.feature.chat.interfaces.EaseChatExtendMenuItemClickListener
import com.hyphenate.easeui.interfaces.OnItemClickListener
import com.hyphenate.easeui.model.EaseMenuItem

class EaseChatExtendMenuAdapter(
    private val isHorizontal: Boolean = false
) :
    EaseBaseChatExtendMenuAdapter<EaseChatExtendMenuAdapter.ViewHolder?, EaseMenuItem?>() {
    private var itemListener: OnItemClickListener? = null
    private var chatMenuItemClickListener: EaseChatExtendMenuItemClickListener? = null
    override val itemLayoutId: Int
        protected get() = if (isHorizontal) R.layout.ease_chat_menu_item_horizontal else R.layout.ease_chat_menu_item

    override fun easeCreateViewHolder(view: View?): ViewHolder {
        return ViewHolder(
            view!!
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        mData?.let {
            if (position >= it.size) return
            it[position]?.run {
                if (resourceId != -1) holder.imageView.setImageResource(resourceId)
                holder.textView.text = title
                if (titleColor != -1){
                    holder.textView.setTextColor(titleColor)
                }
                if (resourceTintColor != -1) {
                    holder.imageView.imageTintList = ColorStateList.valueOf(resourceTintColor)
                }
                holder.itemView.setOnClickListener { view ->
                    chatMenuItemClickListener?.onChatExtendMenuItemClick(menuId, view)
                    itemListener?.onItemClick(view, position)
                }
            }

        }
    }

    /**
     * Set item click listener.
     */
    fun setOnItemClickListener(listener: OnItemClickListener?) {
        itemListener = listener
    }

    /**
     * Set chat extend menu item click listener.
     */
    fun setEaseChatExtendMenuItemClickListener(listener: EaseChatExtendMenuItemClickListener?) {
        chatMenuItemClickListener = listener
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imageView: ImageView
        val textView: TextView

        init {
            imageView = itemView.findViewById<View>(R.id.image) as AppCompatImageView
            textView = itemView.findViewById<View>(R.id.text) as TextView
        }
    }
}